tchannel/testing/vcr/record_modes.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2015 Uber Technologies, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+from .exceptions import VCRError
+
+__all__ = ['RecordMode']
+
+
+class _RecordMode(object):
+    """A _RecordMode dictates a cassette's behavior on recording and replays.
+
+    :param name:
+        Name of the record mode.
+    :param replayable:
+        Whether this record mode allows requests to be replayed.
+    :param can_record:
+        Whether this record mode allows new interactions to be recorded. This
+        may be a boolean or a function that accepts the cassette and returns a
+        boolean.
+    """
+
+    __slots__ = ('name', 'replayable', 'can_record')
+
+    def __init__(self, name, replayable, can_record):
+        self.name = name
+        self.replayable = replayable
+        if not callable(can_record):
+            self.can_record = (lambda _: can_record)
+        else:
+            self.can_record = can_record
+
+
+class RecordMode(object):
+    """
+    Record modes dictate how a cassette behaves when interactions are replayed
+    or recorded. The following record modes are supported.
+
+    .. autoattribute:: ONCE
+       :annotation: = 'once'
+
+    .. autoattribute:: NEW_EPISODES
+       :annotation: = 'new_episodes'
+
+    .. autoattribute:: NONE
+       :annotation: = 'none'
+
+    .. autoattribute:: ALL
+       :annotation: = 'all'
+    """
```

Comment:
This is how this looks, BTW.

![image](https://cloud.githubusercontent.com/assets/41730/8884416/56b6a222-320b-11e5-8d2f-12eae905beda.png)

I wanted it to be clear that those strings can be used to refer to these record modes in addition to `RecordMode.FOO'.